### ISSUE TITLE
Reboot module fails by timeout

### ISSUE THREAD

**Issue by aajisaka** [ansible/awx#5016]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
Reboot module timeouts from AWX, but the same playbook run sucessfully from CLI.

##### ENVIRONMENT
* AWX version: 6.1.0
* AWX install method: docker compose
* Ansible version: 2.8.5 (Python version is upgraded to 3.6.8)
* Operating System: CentOS 7
* Web Browser: Google Chrome

##### STEPS TO REPRODUCE

Run a simple playbook from AWX as follows:
```
- name: reboot
  reboot:
```

##### EXPECTED RESULTS

The playbook ran successfully without timeout.

##### ACTUAL RESULTS

The playbook fails by timeout.
```
TASK [XXX : reboot] ************************************************************
fatal: [XXX]: FAILED! => {"changed": false, "elapsed": 608, "msg": "Timed out waiting for last boot time check (timeout=600)", "rebooted": true}
```

##### ADDITIONAL INFORMATION

When the log level is debug, the following log is displayed many times:
```
reboot: attempting to get system boot time
<XXX> ESTABLISH SSH CONNECTION FOR USER: XXX
<XXX> SSH: EXEC ssh -o ControlMaster=auto -o ControlPersist=10m -o ControlPath=/tmp/ansible-ssh-%h-%p-%r -o ConnectionAttempts=3 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="XXX"' -o ConnectTimeout=10 -tt XXX '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-uzfkugngjniiaztnoyghqeixtehheusb ; cat /proc/sys/kernel/random/boot_id'"'"'"'"'"'"'"'"' && sleep 0'"'"''
<XXX> (141, b'', b'Shared connection to XXX closed.\\r\\n')
<XXX> Failed to connect to the host via ssh: Shared connection to XXX closed.
```

**Comment by ryanpetrello** [comment]
@aajisaka I'm able to reproduce something similar on a nightly install of AWX managing an inventory that contains a single RHEL8 host:

![image](https://user-images.githubusercontent.com/214912/66947882-aa1b1780-f021-11e9-8fef-4bd3571c1c91.png)

**Comment by ryanpetrello** [comment]
cc @samdoran

**Comment by ryanpetrello** [comment]
Ignore me above, I forgot to `ansible_become: true`:

![image](https://user-images.githubusercontent.com/214912/66948252-565cfe00-f022-11e9-8871-ce0e722b4563.png)

**Comment by ryanpetrello** [comment]
Yea, @aajisaka I'm having trouble reproducing this on the most recent version of AWX:

![image](https://user-images.githubusercontent.com/214912/66948323-7d1b3480-f022-11e9-9718-2a1e002b2fc6.png)

Are you sure your hosts aren't just periodically taking too long to reboot?

Given your description above (and verbose output), I'm not certain this is an AWX bug.  What OS are your managed hosts running (AWX shouldn't care, it's just forking an `ansible-playbook` process)?

**Comment by aajisaka** [comment]
> Are you sure your hosts aren't just periodically taking too long to reboot?

Yes, it takes only < 1 min to reboot.

> What OS are your managed hosts running

CentOS 7.7.

**Comment by bluikko** [comment]
I seem to have same issue on CentOS 7 and python 3. The Ansible user is logged in to managed server (the successful login from AWX can be seen in /var/log/secure) but the play is just sitting there waiting until the end of timeout (edit: this is without a lot of verbosity so cannot say if the play is doing something, will test to increase -v)

**Comment by Sprinterfreak** [comment]
Ran into the same issue today. debian-jessie inside lxc container on Proxmox VE

[verbose.txt](https://github.com/ansible/awx/files/4772823/verbose.txt)

I think the reboot module waits for the kernel random/boot_id to change, wich does not happen in containers. Because it is the host kernel's boot_id, right? I think it would be really bad if the container could manage to reboot the host. :)

**Comment by Sprinterfreak** [comment]
This depends on: https://github.com/ansible/ansible/issues/57768 (wich is closed but not fixed)
Corresponding LXC Issue: https://github.com/lxc/lxc/issues/3027

ansible devel-branch release seem to introduce a new option 
```
reboot:
  boot_time_command: 'who -b'
```
Wich would make ansible reboot inside containers work.

I think for the next dacade, we have to stick with `ignore_errors: yes` and fingers crossed.

**Comment by bluikko** [comment]
> I think for the next dacade, we have to stick with `ignore_errors: yes` and fingers crossed.

Ha ha, this would be a really funny comment unless it would be so sad - no, *scandalous*, what is the situation with Ansible and fixing showstopping bugs.

I have started to recommend Salt to everyone starting with automation who could not fix bugs on their own. In my experience they are **far** more responsive to legitimate bug reports from users.

**Comment by samdoran** [comment]
`boot_time_command` was added in Ansible 2.10 to address this, giving maximum flexibility to the playbook author.

**Comment by bchristianv** [comment]
I can confirm this problem still exists in AWX 19.5.1, as described by @aajisaka and @bluikko . What's interesting is that the task status contains `"changed": false` and `"rebooted": true`. 

FWIW, I've run the AWX failing playbook locally, via the execution environment container used by the AWX job template, and it completes successfully.

**Comment by Cloud-Kid** [comment]
I had the same problem I just resolved, my user needed a sudo password to execute this command successfully, by removing the need to enter a sudo password when elevating, the reboot detection worked as intended.

You can do the same by :
- Editing /etc/sudoers or adding a new entry in /etc/sudoers.d/**thenameyouwanttogivetoyourentry** (Whatever you want)
- Adding : **youruser** ALL=(ALL) NOPASSWD:ALL

TLDR : Reboot was allowed without sudo rights but the way Ansible is checking if the reboot succeed wasn't.

Hope this will help if people keep having this issue.
